src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -110,12 +112,82 @@ public Result getLatestResult() {
         return null;
     }
 
+    /**
+     * Used to get result by correction round.
+     * Pay attention, that this method only works for Text, Modeling and FileUpload Exercises!
+     * Programming Submissions can have an arbitrary amount automatic results as well!
+     *
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRound(Long correctionRound) {
```

Comment:
I don't like to have a method that only works for certain exercise types. Can we not make sure this method works for all exercise types? So we first filter out automatic results in a temporary list and then return the correct value?

So basically we use the implementation of `getResultByCorrectionRoundIgnoreAutomatic` but we name it `getResultByCorrectionRound`

Reply:
True, that would be better. I changed it :)